care/facility/models/patient.py

```diff
@@ -683,3 +687,18 @@ class PatientNotes(FacilityBaseModel, PatientRelatedPermissionMixin):
         null=True,
     )
     note = models.TextField(default="", blank=True)
+
+    def save(self, *args, **kwargs):
+        if self.pk:  # Updating an existing note
+            if not self.patient.is_active:
+                raise ValidationError(
+                    {
+                        "patient": "Updating patient data is only allowed for active patients"
+                    }
+                )
+            if now() > self.created_date + datetime.timedelta(
+                seconds=PATIENT_NOTE_EDIT_WINDOW
+            ):
+                raise ValidationError({"note": "Note is not editable anymore"})
```

Comment:
this will prevent updating from the admin page as well,
there might be some cases where we have to fix the records
cc @vigneshhari

Reply:
Do you suggest allowing superuser to bypass this restriction?

Reply:
yes something like that